scripts/linters/pylint_extensions.py

```diff
@@ -1960,6 +2079,388 @@ def process_tokens(self, tokens):
+                multi-line import or single-line import is present, or None
+                if the given exceptional_type is not imported in the module.
+            token_type: int. The kind of token that pylint provided.
+            token: str. The token of module the pylint provided.
+            line: str. The line of the module where current token is present.
+            line_num: int. The line number of given token.
+            exceptional_type: str. The exceptional type for which this method
+                is called, Possible values can be 'Any' or 'object'.
+            node: astroid.scoped_nodes.Module. Node to access module content.
+        """
+        # Checking if linters are in argument-section, return-section or
+        # outside of the function signature.
+        # Eg:
+        #       <outside function signature block>
+        #   def func(<argument-section>) -> <return-section>:
+        #       <outside function signature block>.
+        if token_type == tokenize.NAME:
+            if token == 'def':
+                type_status_dict['outside_function_signature_block'] = False
+                type_status_dict['func_def_start_line'] = line_num
+                type_status_dict['outside_args_section'] = False
+
+        if token_type == tokenize.OP:
+            if token == '->':
+                type_status_dict['outside_args_section'] = True
+                type_status_dict['args_section_end_line_num'] = line_num
+            if type_status_dict['outside_args_section'] and token == ':':
+                type_status_dict['outside_function_signature_block'] = True
+
+        # Checking if exceptional_type is present in function definition or not.
+        if token_type == tokenize.NAME and token == exceptional_type:
+            if not type_status_dict['outside_args_section']:
+                type_status_dict['type_present_inside_arg_section'] = True
+            elif (
+                type_status_dict['outside_args_section'] and
+                type_status_dict['args_section_end_line_num'] == line_num
+            ):
+                type_status_dict['type_present_inside_return_section'] = True
+
+        if (
+            type_status_dict['type_present_inside_arg_section'] or
+            type_status_dict['type_present_inside_return_section']
+        ):
+            type_status_dict['type_present_in_function_signature'] = True
+
+        if type_status_dict['outside_function_signature_block']:
+            if type_status_dict['type_present_in_function_signature']:
+                if (
+                    type_status_dict['type_comment_no_longer_pending'] and
+                    type_status_dict['func_def_start_line'] <= (
+                        type_status_dict['type_comment_line_num'] +
+                        ALLOWED_LINES_OF_GAP_IN_COMMENT
+                    )
+                ):
+                    type_status_dict['type_comment_no_longer_pending'] = False
```

Comment:
Isn't it now `True` that the type comment is no longer pending?

Reply:
No - it should be false here because at this point we have encountered the `type: ignore` for comment. And to check the next comment we have to make it False here. 

```python
# Here we use MyPy ignore because ...            # at this point type_comment_no_longer_pending will change it's value from False to true.
var_1 = 'asd'   # type: ignore[some-msg]         # at this point type_comment_no_longer_pending will change it's value from true to false, so that in next line if linters encounter new comment then it can be True again.


# Here we use MyPy ignore because ...        # Here again type_comment_no_longer_pending will change it's value from false to True and it will only become false when linter encounter `type: ignore`.

```

Reply:
I agree that your implementation is technically correct. I'm arguing that the name of the dictionary key is misleading. To take your example:

```
# Here we use MyPy ignore because ...            # At this point the type comment is now pending because we're looking for the associated type: ignore comment
var_1 = 'asd'   # type: ignore[some-msg]         # Now we've found the type: ignore comment, so the comment we found above is no longer pending
```

In other words, I think the key should be `type_comment_pending`

Reply:
Fixed.